### PR TITLE
DOC/CODESTYLE Add code style example

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
## What

Code style examples

## Why ?

Code style example may help for developing OpenUCX.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: doc/CodeStyle.md [hunk 1]
```diff
@@ -63,3 +63,100 @@
...
+* ## Examples
+
+* ### if style
+
+Good
+
+```C
+    if (val != XXX) {
+        /* snip */
+    } else if (val == YYY) {}
```
Comment (yosefe) [inline doc/CodeStyle.md]: redundant } ?

**Comment by yosefe** [comment]
ok to test

**Comment by yosefe** [comment]
@shamisp @tonycurtis pls take a look

File: doc/CodeStyle.md (hunk 1 above)
Comment (shamisp) [reply doc/CodeStyle.md]: for single liner ?

**Comment by shamisp** [comment]
> @shamisp @tonycurtis pls take a look

@tonycurtis  is OOO so it is just me for this week , unless Tony want to undo OOO :)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9593/ for details (Mellanox internal link).

**Pull request merged**